lib/Dialect/Moore/MooreOps.cpp

```diff
@@ -287,6 +287,30 @@ VariableOp::handlePromotionComplete(const MemorySlot &slot, Value defaultValue,
 
 LogicalResult VariableOp::canonicalize(VariableOp op,
                                        ::mlir::PatternRewriter &rewriter) {
+  if (auto initial = op.getInitial()) {
+    return TypeSwitch<Type, LogicalResult>(op.getType().getNestedType())
+        .Case<StructType, UnpackedStructType>([&op, &rewriter,
+                                               &initial](auto &type) {
+          SmallVector<Value> createFields;
+          unsigned bit = 0;
+          auto i32 = moore::IntType::getInt(op->getContext(), 32);
+          for (const auto &member : type.getMembers()) {
+            auto memberType = cast<UnpackedType>(member.type);
+            if (auto width = memberType.getBitSize()) {
+              auto lowBit = rewriter.create<ConstantOp>(op.getLoc(), i32, bit);
+              bit += width.value();
+              auto field = rewriter.create<ExtractOp>(op->getLoc(), member.type,
```

Comment:
I think UnpackedStructType doesn't allow bit extraction. Is it expected way to get a field from UnpackedStructType?

Reply:
Yes it should use `structExtractOp` to extract a struct type.